sources/lib/DependencyInjection/PommExtension.php

```diff
@@ -60,11 +62,39 @@ public function configure(array $configs, ContainerBuilder $container)
         $configuration = new Configuration();
         $config = $processor->processConfiguration($configuration, $configs);
 
-        $container->setParameter('pomm.configuration', $config['configuration']);
+        $definition = $container->getDefinition('pomm');
+        foreach ($config['configuration'] as $name => $pommConfig) {
+            if (isset($pommConfig['session_builder'])) {
+                $service = $container->getDefinition($pommConfig['session_builder']);
+                $service->setArguments([$pommConfig]);
+
+                $definition->addMethodCall('addBuilder', [$name, new Reference($pommConfig['session_builder'])]);
+            } else {
+                $service = uniqid($pommConfig['class:session_builder'], true);
+                $cbDefinition = $container->register($service, ltrim($pommConfig['class:session_builder'], '\\'));
+                $cbDefinition->setShared(false);
+                $cbDefinition->setArguments([$pommConfig]);
+
+                $definition->addMethodCall('addBuilder', [$name, new Reference($service)]);
+            }
+
+            if (isset($pommConfig['pomm:default']) && $pommConfig['pomm:default']) {
+                $definition->addMethodCall('setDefaultBuilder', [$name]);
+
+                $container->setAlias('pomm.default_session', sprintf('pomm.%s_session', $name));
+            }
+
+            //register all session's into the container
+            $session = new Definition('PommProject\Foundation\Session\Session');
+            $session->setFactory([new Reference('pomm'), 'getSession'])
+                ->addArgument($name)
+                ;
+            $container->addDefinitions([sprintf('pomm.%s_session', $name) => $session]);
```

Comment:
The default one is singular. And service is also singular e.g one instance by default.